aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -871,13 +896,20 @@ func resourceAviatrixTransitGatewayCreate(d *schema.ResourceData, meta interface
 		gateway.Interfaces = b64.StdEncoding.EncodeToString(interfaceList)
 
 		if goaviatrix.IsCloudType(cloudType, goaviatrix.EDGENEO) {
-			// set the static interface mapping for AEP
-			interfaceMapping := map[string][]string{
-				"eth0": {"wan", "0"},
-				"eth1": {"wan", "1"},
-				"eth2": {"wan", "2"},
-				"eth3": {"mgmt", "0"},
-				"eth4": {"wan", "3"},
+			interfaceMappingInput := d.Get("interface_mapping").([]interface{})
+			interfaceMapping := make(map[string][]string)
+			for _, value := range interfaceMappingInput {
+				mappingMap, ok := value.(map[string]interface{})
+				if !ok {
+					return fmt.Errorf("invalid type for interface mapping, expected a map")
+				}
+				interfaceName, ok1 := mappingMap["name"].(string)
+				interfaceType, ok2 := mappingMap["type"].(string)
+				interfaceIndex, ok3 := mappingMap["index"].(string)
+				if !ok1 || !ok2 || !ok3 {
+					return fmt.Errorf("invalid interface mapping, 'name', 'type', and 'index' must be strings")
+				}
+				interfaceMapping[interfaceName] = []string{interfaceType, interfaceIndex}
```

Comment:
What happens if `interfaceName` is not unique? What happens if `interfaceIndex` is not unique?

Reply:
The backend api handles the validation of interface mapping.